Tests/Foundation/Tests/TestThread.swift

```diff
@@ -64,57 +45,36 @@ class TestThread : XCTestCase {
         XCTAssertTrue(ok, "NSCondition wait timed out")
     }
     
-#if NS_FOUNDATION_ALLOWS_TESTABLE_IMPORT
     func test_threadName() {
+
+        func testInternalThreadName(_ name: String?) {
+#if NS_FOUNDATION_ALLOWS_TESTABLE_IMPORT
+            XCTAssertEqual(Thread.current._name, name)
+#endif
+        }
+
 #if os(Linux) || os(Android) // Linux sets the initial thread name to the process name.
         XCTAssertEqual(Thread.current.name, "TestFoundation")
-        XCTAssertEqual(Thread.current._name, "TestFoundation")
+        testInternalThreadName("TestFoundation")
 #else
         // No name is set initially
         XCTAssertEqual(Thread.current.name, "")
-        XCTAssertEqual(Thread.current._name, "")
+        testInternalThreadName("")
 #endif
         Thread.current.name = "mainThread"
         XCTAssertEqual(Thread.mainThread.name, "mainThread")
-        XCTAssertEqual(Thread.mainThread._name, "mainThread")
-
-        let condition = NSCondition()
-        condition.lock()
-
-        let thread2 = Thread() {
-            XCTAssertEqual(Thread.current.name, "Thread2-1")
-
-            Thread.current.name = "Thread2-2"
-            XCTAssertEqual(Thread.current.name, "Thread2-2")
-            XCTAssertEqual(Thread.current._name, Thread.current.name)
-
-            Thread.current.name = "12345678901234567890"
-            XCTAssertEqual(Thread.current.name, "12345678901234567890")
-#if os(macOS) || os(iOS)
-            XCTAssertEqual(Thread.current._name, Thread.current.name)
-#elseif os(Linux) || os(Android)
-            // pthread_setname_np() only allows 15 characters on Linux, so setting it fails
-            // and the previous name will still be there.
-            XCTAssertEqual(Thread.current._name, "Thread2-2")
-#endif
-            condition.lock()
-            condition.signal()
-            condition.unlock()
-        }
-        thread2.name = "Thread2-1"
-        thread2.start()
-
-        // Allow 1 second for thread2 to finish
-        XCTAssertTrue(condition.wait(until: Date(timeIntervalSinceNow: 1)))
-        condition.unlock()
+        testInternalThreadName("mainThread")
```

Comment:
This might be something to address in another pr, but since thread names aren't supported on TARGET_OS_BSD (that is, `pthread_setname_np` nor `pthread_getname_np` are available), these tests will trip expectedly. Maybe the #if clauses can be tweaked?

Reply:
I think for targets that dont support `pthread_{get,set}name_np`, the name can just be stored in a `String` in the `Thread` object.

Do the `swift-corelibs-foundation` tests build & run at all on BSD and is this going to break them or is it just another thing to remember to fix when the tests are ported?

Reply:
If it's as simple as a string variable in the other cases, then I think we can leave it as "another thing to remember to fix when the tests are ported". 

(There's still more hurdles to jump through on the porting -- like Dispatch and friends, _especially_ -- but with the WIP Swift BSD port, a decent percentage of Foundation tests appear to be running and passing.)